src/components/NoteForm.vue

```diff
@@ -1,74 +1,54 @@
-<script setup>
-import { ref } from 'vue'
-import { defineProps } from 'vue';
+<script setup lang="ts">
+import { ref } from 'vue';
+import { Button } from '@/components/ui/button'
 
-const props = defineProps({
-  isInputActive: Boolean,
-  openInput: Function
-})
-// const title = ref('');
-// const content = ref('');
-const notes = ref([]);
+interface Note {
+  title: string;
+  content: string;
+}
+
+const emits = defineEmits(["updateNotes"])
+const { isInputActive, openInput, closeInput } = defineProps(['isInputActive', 'openInput', 'closeInput']);
 
-const note = ref({
-  title: '',
-  content: ''
-})
+const notes = ref<Note[]>([]);
+const default_note: Note = { title: '', content: '' };
+const note = ref<Note>({ ...default_note });
 
 const handleSubmit = () => {
-  notes.value = [...notes.value, note.value]
-  console.log(notes.value)
-  note.value = { title: '', content: '' };
+  if (isNoteValid(note.value.title, note.value.content)) {
+    notes.value = [...notes.value, note.value];
+    emits('updateNotes', notes.value)
+    closeInput()
+    resetForm()
+  } else {
+    console.log('Tous les champs ne sont pas renseignés')
+  }
+};
+
+const isNoteValid = (title: string, content: string): boolean => {
+  return title.trim() !== '' && content.trim() !== '';
```

Comment:
```suggestion
  return !!(title?.trim() && content?.trim());
```
Comme ça tu couvres aussi le `null` et le `undefined`.